community/server/src/main/java/org/neo4j/server/rest/dbms/AuthorizationEnabledFilter.java

```diff
@@ -125,13 +131,22 @@ public void doFilter( ServletRequest servletRequest, ServletResponse servletResp
             default:
                 log.warn( "Failed authentication attempt for '%s' from %s", username, request.getRemoteAddr() );
                 requestAuthentication( request, invalidCredential ).accept( response );
+            }
+        }
+        catch ( InvalidAuthTokenException e )
+        {
+            requestAuthentication( request, invalidAuthToken( e.getMessage() ) ).accept( response );
         }
     }
 
-    private AuthSubject authenticate( String username, String password )
+    private AuthSubject authenticate( String username, String password ) throws InvalidAuthTokenException
     {
         AuthManager authManager = authManagerSupplier.get();
-        AuthSubject authSubject = authManager.login( username, password );
+
+        Map<String,Object> authToken = map( AuthToken.SCHEME_KEY,
```

Comment:
I guess we could add a newBasicAuthToken() factory method. But this would be the only production code location where we create an auth token on the server side.